hyperspy/signal.py

```diff
@@ -3111,6 +3112,83 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            replace_axis_index,
```

Comment:
```suggestion
                            replace_axis_index = 0,
```

Should have a default value so that when signal has only one axis, one does not need to specify the index.

In other methods (e.g. stack, split) this parameter is just called `axis`, which might be ambiguous though. We should however try to be consistent across different methods. In the `set_axis` method it is called `index_in_axes_manager`.

Reply:
Yes, I was unsure how to name this parameter, because there are 2 axes and hence I didn't want to call it `axis`. I changed it to `index_in_axes_manager` now, because it suggests that this parameter specifies the axis that will be replaced.